library/src/main/java/com/chuckerteam/chucker/internal/ui/transaction/TransactionPayloadFragment.kt

```diff
@@ -293,26 +293,23 @@ internal class TransactionPayloadFragment :
     }
 
     @Suppress("ThrowsCount")
-    private suspend fun saveToFile(type: Int, uri: Uri, transaction: HttpTransaction): Boolean {
+    private suspend fun saveToFile(type: PayloadType, uri: Uri, transaction: HttpTransaction): Boolean {
         return withContext(Dispatchers.IO) {
             try {
                 requireContext().contentResolver.openFileDescriptor(uri, "w")?.use {
                     FileOutputStream(it.fileDescriptor).use { fos ->
                         when (type) {
-                            TYPE_REQUEST -> {
+                            is PayloadType.Request -> {
                                 transaction.requestBody?.byteInputStream()?.copyTo(fos)
                                     ?: throw IOException(TRANSACTION_EXCEPTION)
                             }
-                            TYPE_RESPONSE -> {
+                            is PayloadType.Response -> {
                                 transaction.responseBody?.byteInputStream()?.copyTo(fos)
-                                    ?: throw IOException(TRANSACTION_EXCEPTION)
-                            }
-                            else -> {
-                                if (transaction.responseImageData != null) {
-                                    fos.write(transaction.responseImageData)
-                                } else {
-                                    throw IOException(TRANSACTION_EXCEPTION)
-                                }
+                                    ?: if (transaction.responseImageData != null) {
+                                        fos.write(transaction.responseImageData)
+                                    } else {
+                                        throw IOException(TRANSACTION_EXCEPTION)
+                                    }
```

Comment:
This should be rewritten in just:

```
transaction.responseBody?.byteInputStream()?.copyTo(fos)
    ?: throw IOException(TRANSACTION_EXCEPTION)	                                   
```

Reply:
Note that with this change we are going to get errors every time user tries to save response with image. Now it saves it, despite not opening it due no file extension.

I wanted to address it in a separate issue to have a clear file type for saved transactions, like `txt` for text transactions, so users could open this saved file easily right on their devices.

Reply:
Changing as you are suggesting, since it is useless anyway to save an image without being able to open it with ease later.